src/include/simeng/CoreInstance.hh

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <string>
+
+#include "simeng/AlwaysNotTakenPredictor.hh"
+#include "simeng/Core.hh"
+#include "simeng/Elf.hh"
+#include "simeng/FixedLatencyMemoryInterface.hh"
+#include "simeng/FlatMemoryInterface.hh"
+#include "simeng/GenericPredictor.hh"
+#include "simeng/ModelConfig.hh"
+#include "simeng/SpecialFileDirGen.hh"
+#include "simeng/arch/Architecture.hh"
+#include "simeng/arch/aarch64/Architecture.hh"
+#include "simeng/arch/aarch64/Instruction.hh"
+#include "simeng/arch/aarch64/MicroDecoder.hh"
+#include "simeng/kernel/Linux.hh"
+#include "simeng/models/emulation/Core.hh"
+#include "simeng/models/inorder/Core.hh"
+#include "simeng/models/outoforder/Core.hh"
+#include "simeng/pipeline/A64FXPortAllocator.hh"
+#include "simeng/pipeline/BalancedPortAllocator.hh"
+#include "yaml-cpp/yaml.h"
+
+namespace simeng {
+
+enum class SimulationMode { Emulation, InOrderPipelined, OutOfOrder };
+
+/** A class to create a SimEng core instance from a supplied config. */
+class CoreInstance {
+ public:
+  /** Default constructor with a no passed configuration. */
+  CoreInstance();
+
+  /** Constructor with a config file path. */
+  CoreInstance(std::string configPath);
+
+  /** Constructor with a pre-constructed config file. */
+  CoreInstance(YAML::Node config);
```

Comment:
What is the use case for this constructor? If the configPath is provided then CoreInstance::CoreInstance(std::string configPath) will handle that case. If it isn't the default config is loaded in CoreInstance::CoreInstance(). Is this constructor used anywhere?

I'm not sure about this, so please correct me if I'm wrong but wouldn't declaring a public constructor with the YAML::Node argument make it necessary for every library/executable which uses CoreInstance to also link yaml-cpp explicitly? In which case it would still require us to link yaml-cpp to the SST build?

Reply:
The comments made are correct. Forgot to push commit which removed this constructor. It was initially used by the regression test suite but I found it was better to forgo the use of the CoreInstance class and use the prior core creation method.